src/common/snippets/src/lowered/expression_factory.cpp

```diff
@@ -91,7 +91,7 @@ ExpressionPtr LinearIR::ExpressionFactory::create(const std::shared_ptr<op::Loop
                                                   const std::vector<PortConnectorPtr>& inputs,
                                                   const LinearIR& linear_ir) {
     OPENVINO_ASSERT(inputs.empty(), "LoopBegin cannot have inputs");
-    auto expr = std::make_shared<Expression>(Expression(n, linear_ir.m_shape_infer_factory));
+    auto expr = std::shared_ptr<Expression>(new Expression(n, linear_ir.m_shape_infer_factory));
```

Comment:
What is the reason for this change? We can't use `make_shared` for expressions anymore?

Reply:
The reason is that we implemented a [protected copy constructor](https://github.com/IvanNovoselov/openvino/blob/1e43b55755396b62f3227b9a33860f2a787fdefa/src/common/snippets/include/snippets/lowered/expression.hpp#L64) instead of a default public one.

Reply:
Thanks for the explanation